### PR TITLE
Update Finetuner hyperparameters

### PR DESCRIPTION
**Pull request by lucas-aixplain**
- Add `train_batch_size` and `eval_batch_size` as available parameters for fine-tuning LLMs.
- Add constraints to limit possible values based on conducted tests.
- Add unit test for hyperparameters.

Task: M-6021007588

### CODE REVIEW

File: aixplain/modules/finetune/hyperparameters.py [hunk 1]
```diff
@@ -56,6 +65,12 @@ def __post_init__(self):
         if self.epochs > EPOCHS_MAX_VALUE:
             raise ValueError(f"epochs must be one less than {EPOCHS_MAX_VALUE}")
```
Comment (thiago-aixplain) [inline aixplain/modules/finetune/hyperparameters.py:66]: less or equal

File: aixplain/modules/finetune/hyperparameters.py [hunk 2]
```diff
@@ -56,6 +65,12 @@ def __post_init__(self):
...
 
+        if self.train_batch_size not in BATCH_SIZE_VALUES:
+            raise ValueError(f"train_batch_size must be one of the following values: {BATCH_SIZE_VALUES}")
+
+        if self.eval_batch_size not in BATCH_SIZE_VALUES:
+            raise ValueError(f"eval_batch_size must be one of the following values: {BATCH_SIZE_VALUES}")
+
         if self.max_seq_length > MAX_SEQ_LENGTH_MAX_VALUE:
```
Comment (thiago-aixplain) [inline aixplain/modules/finetune/hyperparameters.py:74]: less or equal

**Review by thiago-aixplain — APPROVED** [review]

File: aixplain/modules/finetune/hyperparameters.py (hunk 1 above)
Comment (lucas-aixplain) [reply aixplain/modules/finetune/hyperparameters.py:66]: In this case, we should only raise ValueError if the number of epochs is greater than the maximum allowed value (`EPOCHS_MAX_VALUE`), so the logic is correct.

File: aixplain/modules/finetune/hyperparameters.py (hunk 2 above)
Comment (lucas-aixplain) [reply aixplain/modules/finetune/hyperparameters.py:74]: For batch size, I want only to receive values from the list BATCH_SIZE_VALUES, that's why we have `not in` there.

File: aixplain/modules/finetune/hyperparameters.py (hunk 1 above)
Comment (lucas-aixplain) [reply aixplain/modules/finetune/hyperparameters.py:66]: I'll change the message to ensure it's correct.

**Pull request merged**